### PR TITLE
add ssl

### PR DESCRIPTION
**Pull request by aiwantaozi**
add ssl support for logging target elasticsearch, splunk and kafka

### CODE REVIEW

File: apis/management.cattle.io/v3/logging_types.go [hunk 1]
```diff
@@ -94,19 +94,27 @@ type LoggingCondition struct {
...
+	Endpoint      string `json:"endpoint,omitempty" norman:"required"`
+	IndexPrefix   string `json:"indexPrefix,omitempty" norman:"required"`
+	DateFormat    string `json:"dateFormat,omitempty" norman:"required,type=enum,options=YYYY-MM-DD|YYYY-MM|YYYY,default=YYYY-MM-DD"`
+	AuthUserName  string `json:"authUsername,omitempty"` //secret
+	AuthPassword  string `json:"authPassword,omitempty"` //secret
+	Certificate   string `json:"certificate"`
+	ClientCert    string `json:"clientCert"`
+	ClientKey     string `json:"clientKey"`
+	ClientKeyPass string `json:"clientKeyPass"`
+	SSLVerify     bool   `json:"sslVerify" norman:"default=false"`
```
Comment (alena1108) [inline apis/management.cattle.io/v3/logging_types.go]: default is always false for the boolean, so there is no need to define the norman tag

File: apis/management.cattle.io/v3/logging_types.go [hunk 2]
```diff
@@ -124,6 +132,9 @@ type KafkaConfig struct {
 	ZookeeperEndpoint string   `json:"zookeeperEndpoint,omitempty"`
```
Comment (alena1108) [inline apis/management.cattle.io/v3/logging_types.go]: same here, no need to define the default

File: apis/management.cattle.io/v3/logging_types.go [hunk 3]
```diff
@@ -124,6 +132,9 @@ type KafkaConfig struct {
...
 	BrokerEndpoints   []string `json:"brokerEndpoints,omitempty"`
 	Topic             string   `json:"topic,omitempty" norman:"required"`
+	Certificate       string   `json:"certificate"`
+	ClientCert        string   `json:"clientCert"`
+	ClientKey         string   `json:"clientKey"`
 }
```
Comment (alena1108) [inline apis/management.cattle.io/v3/logging_types.go]: no SSLVerify for Zookeeper? Just wanted to make sure it's intended, as other configs have this flag set.

Comment (aiwantaozi) [reply apis/management.cattle.io/v3/logging_types.go]: Fluentd Kafka plugin does not support sslVerify yet, need to send PR to them, so not in this time

**Pull request merged**